src/client/creature.cpp

```diff
@@ -510,19 +510,17 @@ void Creature::updateWalkAnimation(int totalPixelsWalked)
     if(m_outfit.getCategory() != ThingCategoryCreature)
         return;
 
-    int footAnimPhases = getAnimationPhases() - 1;
-    int footDelay = getStepDuration(true) / getAnimationPhases() + 15;
+    int footAnimPhases, footDelay;
     if (m_outfit.getMount() != 0) {
+        // Since mount is a different outfit we need to get the mount animation phases
         auto datType = g_things.rawGetThingType(m_outfit.getMount(), ThingCategoryCreature);
-        footAnimPhases = datType->getAnimationPhases() - 1;
-        if (datType->getIdleAnimator() != nullptr) {
-            footDelay = getStepDuration(true) / ceil(footAnimPhases / 2) + 15;
-        }
-    } else if (getIdleAnimator() != nullptr) {
-        footDelay = getStepDuration(true) / ceil(footAnimPhases / 2) + 15;
+        footAnimPhases = datType->getAnimationPhases();
+    } else {
+        footAnimPhases = getAnimationPhases();
     }
+    footAnimPhases = std::max<int>(0, footAnimPhases - 1);
```

Comment:
I personally don't like separating declaration and definition of variables, so I would opt for a ternary expression:

```cpp
int footAnimPhases = std::max<int>(0, (
  m_outfit.getMount()
    ? g_things.rawGetThingType(m_outfit.getMount(), ThingCategoryCreature)->getAnimationPhases()
    : getAnimationPhases()) - 1
); 
```

However feel free to dismiss it :wink:

Reply:
Well, I also like declaration + definition… But for me in this case your code(and anything that I come up with) don't look simpler.